test/integration/targets/openssl_certificate_info/tasks/main.yml

```diff
@@ -0,0 +1,151 @@
+      postalCode: "1234"
+    useCommonNameForSAN: no
+    key_usage:
+      - digitalSignature
+      - keyAgreement
+      - Non Repudiation
+      - Key Encipherment
+      - dataEncipherment
+      - Certificate Sign
+      - cRLSign
+      - Encipher Only
+      - decipherOnly
+    key_usage_critical: yes
+    extended_key_usage:
+      - serverAuth  # the same as "TLS Web Server Authentication"
+      - TLS Web Server Authentication
+      - TLS Web Client Authentication
+      - Code Signing
+      - E-mail Protection
+      - timeStamping
+      - OCSPSigning
+      - Any Extended Key Usage
+      - qcStatements
+      - DVCS
+      - IPSec User
+      - biometricInfo
+    subject_alt_name:
+      - "DNS:www.ansible.com"
+      - "IP:1.2.3.4"
+      - "IP:::1"
+      - "email:test@example.org"
+      - "URI:https://example.org/test/index.html"
+    basic_constraints:
+      - "CA:TRUE"
+      - "pathlen:23"
+    basic_constraints_critical: yes
+    ocsp_must_staple: yes
+
+- name: Generate CSR 2
+  openssl_csr:
+    path: '{{ output_dir }}/csr_2.csr'
+    privatekey_path: '{{ output_dir }}/privatekeypw.pem'
+    privatekey_passphrase: hunter2
+    useCommonNameForSAN: no
+    basic_constraints:
+      - "CA:TRUE"
+
+- name: Generate CSR 3
+  openssl_csr:
+    path: '{{ output_dir }}/csr_3.csr'
+    privatekey_path: '{{ output_dir }}/privatekey.pem'
+    useCommonNameForSAN: no
+    subject_alt_name:
+      - "DNS:*.ansible.com"
+      - "DNS:*.example.org"
+      - "IP:DEAD:BEEF::1"
+    basic_constraints:
+      - "CA:FALSE"
+
+- name: Generate CSR 4
+  openssl_csr:
+    path: '{{ output_dir }}/csr_4.csr'
+    privatekey_path: '{{ output_dir }}/privatekey.pem'
+    useCommonNameForSAN: no
+
+- name: Generate selfsigned certificates
+  openssl_certificate:
+    path: '{{ output_dir }}/cert_{{ item }}.pem'
+    csr_path: '{{ output_dir }}/csr_{{ item }}.csr'
+    privatekey_path: '{{ output_dir }}/privatekey.pem'
+    provider: selfsigned
+    selfsigned_digest: sha256
+    selfsigned_not_after: "+10d"
+    selfsigned_not_before: "-3d"
+  loop:
+  - 1
+  - 2
+  - 3
+  - 4
+
+- name: Prepare result list
+  set_fact:
+    info_results: []
+
+- name: Running tests with pyOpenSSL backend
+  include_tasks: impl.yml
+  vars:
+    select_crypto_backend: pyopenssl
+  when: pyopenssl_version.stdout is version('0.15', '>=')
+
+- name: Prepare result list
+  set_fact:
+    pyopenssl_info_results: "{{ info_results }}"
+    info_results: []
+
+- name: Running tests with cryptography backend
+  include_tasks: impl.yml
+  vars:
+    select_crypto_backend: cryptography
+  when: cryptography_version.stdout is version('1.6', '>=')
+
+- name: Prepare result list
+  set_fact:
+    cryptography_info_results: "{{ info_results }}"
+
+- block:
+  - name: Dump pyOpenSSL results
+    debug:
+      var: pyopenssl_info_results
+  - name: Dump cryptography results
+    debug:
+      var: cryptography_info_results
+  - name: Compare results
+    assert:
```

Comment:
shouldn't that fail CSR/Cert number 2 later, since it was issued for the private key with a password?

Reply:
The private key isn't used/needed for retrieving information on certificates.

(What's interesting though is that creating the self-signed certificate works with the wrong key... But that's unrelated to this PR...)

Reply:
So this would only be discovered if there's also an `openssl_privatekey_info` task and then the public keys of the subject, issuer and private key need to match up?

Reply:
Passphrases are only discovered when private keys protected with passphrases are loaded, it doesn't matter by whom.

Does the cert actually contain the public key of the issuer? I thougt there's only the signature, which can be validated given the public key of the issuer (which has to be taken fom the issuer's cert, or from another source).

Reply:
https://www.alvestrand.no/objectid/2.5.29.35.html exists, but that's not set in many cases. We'd need #29951 for that.

Reply:
There's also a feature request for that: #50782

But that doesn't store the public key, only some kind of hash, so you still need to get the public key from somewhere else :)

Reply:
Urgh, sometimes x509 really feels like they want to make it as hard as possible to varify/validate information... :-/

So in this case (selfsigned) only `Issuer == Subject` can be checked currently and the matching public key check once the `openssl_privatekey_info` module is here...

Reply:
The aim is also to make the cert small, so you can't insert everything ;) If the objectID extension is there, you can also check if the cert's public key matches it.

Anyway, it might make sense to have another module which does some more validation, or have some more parameters to allow more complex information retrieval. But I wouldn't do that for 2.8, since we have to look at it in some more detail to find a really good solution ;)